src/util/irep_serialization.cpp

```diff
@@ -324,11 +324,15 @@ Function: irep_serializationt::read_gb_string
 
 irep_idt irep_serializationt::read_gb_string(std::istream &in)
 {
-  char c;
   size_t length=0;
 
-  while((c = in.get()) != 0)
+  while(char c = in.get())
   {
+    if(!in.good())
+    {
+      throw irep_serialization_errort("gb string is not null-terminated");
```

Comment:
It's just that VS doesn't understand C++11. Better to fix the exception constructor than to throw a char star though surely? 
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#e14-use-purpose-designed-user-defined-types-as-exceptions-not-built-in-types